idea/src/org/jetbrains/kotlin/idea/inspections/collections/SimplifiableCallChainInspection.kt

```diff
@@ -82,17 +79,10 @@ class SimplifiableCallChainInspection : AbstractCallChainChecker() {
     }
 }
 
-private fun KotlinType?.isMap(builtIns: KotlinBuiltIns): Boolean {
-    if (this == null) return false
-
-    val map = builtIns.map.defaultType
-    val mutableMap = builtIns.mutableMap.defaultType
+private fun KotlinType?.isMapOrSubType(): Boolean {
+    return this != null && (isMap() || constructor.supertypes.any { it.isMap() })
```

Comment:
It's not the implementation that @mglukhikh or me have suggested. Please re-read the discussion. This code would return true on a Java class named `SomethingMap` which does not inherit from `Map`, which is incorrect. What we're looking for is classes whose names both end with `"Map"` _and_ who inherit from `Map`.